src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -258,7 +208,8 @@ private void TryReleaseLeaseIfOwned()
             {
                 if (HasLease)
                 {
-                    _lockBlob.ReleaseLease(new AccessCondition { LeaseId = LeaseId });
+                    Task.Run(() => _lockManager.ReleaseLockAsync(_lockHandle, CancellationToken.None)).GetAwaiter().GetResult();
```

Comment:
We can't - this is called from Dispose, which is non-async.  :(    

---
In reply to: [123671702](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615#discussion_r123671702) [](ancestors = 123671702)